src/main/java/nl/tudelft/controller/ShopStateController.java

```diff
@@ -19,18 +21,26 @@ public ShopStateController(ShopState shopState) {
     }
 
     /**
-     * Checks if the selected player has enough money and an item is selected,
-     * buys the item and aplies it to the selected player.
+     * Checks if the selected player has enough money and an item is selected, buys the item and
+     * aplies it to the selected player.
      */
     public void applyUpgrade() {
         if (shopState.getSelectedItem() != null
-                && shopState.getSelectedItem().getPrice() <= shopState
-                        .getSelectedPlayer().getMoney()) {
+                && shopState.getSelectedItem().getPrice() <= shopState.getShop().getGame()
+                        .getPlayers()[shopState.getSelectedPlayer()].getMoney()) {
 
-            shopState.getSelectedItem().applyTo(shopState.getSelectedPlayer());
-            shopState.getSelectedPlayer().setMoney(
-                    shopState.getSelectedPlayer().getMoney()
+            Player modifiedPlayer =
```

Comment:
This could probably do with a comment explaining what the hell is actually going on.

< documentation